### PR TITLE
add document for auc and imageset

### PR DESCRIPTION
**Pull request by dding3**

### CODE REVIEW

File: zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/keras/metrics/AUC.scala [hunk 1]
```diff
@@ -23,6 +23,16 @@ import com.intel.analytics.bigdl.tensor.TensorNumericMath.TensorNumeric
 
 import scala.collection.mutable.ArrayBuffer
 
+/**
+ * Represent an accuracy result. It's equal to the probability that a
+ * classifier will rank a randomly chosen positive instance higher
+ * than a randomly chosen negative one.
+ * Refer: https://en.wikipedia.org/wiki/Receiver_operating_characteristic
```
Comment (hhbyyh) [inline zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/keras/metrics/AUC.scala]: RoC is not the only AUC metrics, do we plan to support other AUC with the same class?

File: docs/docs/ProgrammingGuide/metrics.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
+ValidationMethod is a method to validate the model during model training or evaluation.
+The trait can be extended by user-defined method. Now we have defined AUC to evaluate model.
+
+---
+## AUC ####
+Area under the ROC curve. More information about ROC can be found https://en.wikipedia.org/wiki/Receiver_operating_characteristic
+It's used to evaluate a binary(0/1 only) classification mode. It supports single label and multiple labels.
```
[the remark this answers is not in the document]
Comment (dding3) [reply docs/docs/ProgrammingGuide/metrics.md]: Yes, with auc, user has to convert their label to 0/1. As auc is for binary classification, I think 0/1 label should be acceptable to users.

CrossEntropyCriterion in BigDL doesn't support 0 as label and it's usually used for multiple classfication. While auc is for binary classification. For other criterion, eg, MultiLabelSoftMarginCriterion, label can be 0 and we can use auc.

File: docs/docs/ProgrammingGuide/metrics.md [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+sc = get_nncontext()
+
+data_len = 4
+batch_size = 8
+FEATURES_DIM = 2
+
+def gen_rand_sample():
+    features = np.random.uniform(0, 1, (FEATURES_DIM))
+    label = np.random.uniform(0, 1, (FEATURES_DIM))
```
[the remark this answers is not in the document]
Comment (dding3) [reply docs/docs/ProgrammingGuide/metrics.md]: Nice catch. Label should be binary here. Thanks for helping me find a bug in AUC :)

File: docs/docs/ProgrammingGuide/metrics.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (hhbyyh) [reply docs/docs/ProgrammingGuide/metrics.md]: OK. Let's leave it as it is. I think the correct approach is to have all the criterion support 0 as label.

File: zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/keras/metrics/AUC.scala (hunk 1 above)
Comment (dding3) [reply zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/keras/metrics/AUC.scala]: We can add them if needed. 

Currently add comments to clarify that it's auc for ROC curve.

File: docs/docs/ProgrammingGuide/metrics.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (dding3) [reply docs/docs/ProgrammingGuide/metrics.md]: I guess CrossEntropyCriterion in bigdl doesn't support 0 as label as we want to behave the same with torch. If there is concern, maybe we can create a issue to BigDL :)

**Comment by hhbyyh** [comment]
LGTM

**Comment by dding3** [comment]
Jekins passed. http://172.168.2.101:8080/job/ZOO-PR-Validation/333/

Thanks for your review.
